cmake/DetectCodeCoverage.cmake

```diff
@@ -20,7 +26,8 @@ if ( GCOV AND
      LCOV AND
```

Comment:
@jbakosi : I think the logic here is wrong, whenever `gcov`, `lcov`, `genhtml` and `sed` are found, coverage is enabled. It seems to me this block should be wrapped in some like `if(ENABLE_COVERRAGE)`.